### PR TITLE
chore: adds testcafe

### PR DESCRIPTION
**Pull request by manueltanzi-okta**
### Description

- Adds testcafe framework
- Add some PrimaryAuth page tests
    
Resolves: [OKTA-243016](https://oktainc.atlassian.net/browse/OKTA-243016)

### Video

testcafe_v2

### Reviewers

@haishengwu-okta @nikhilvenkatraman-okta @robertdamphousse-okta

### CODE REVIEW

File: test/testcafe/framework/page-objects/BasePageObject.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export default class BasePageObject {
+  constructor(t) {
+    this.t = t;
+  }
+
+  async navigateToPage() {
+    await this.t.navigateTo(`http://localhost:3000${this.url}?hideNav=true`);
```
Comment (haishengwu-okta) [inline test/testcafe/framework/page-objects/BasePageObject.js]: probably dont need `hideNav..`

File: test/testcafe/framework/page-objects/PrimaryAuthPageObject.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import BasePageObject from './BasePageObject';
+import BaseFormObject from './components/BaseFormObject';
+
+export default class PrimaryAuthPageObject extends BasePageObject {
+  constructor (t) {
+    super(t);
+    this.url = '';
```
Comment (haishengwu-okta) [inline test/testcafe/framework/page-objects/PrimaryAuthPageObject.js]: maybe set url to empty by default at parent class?

File: test/testcafe/framework/page-objects/IdentityPageObject.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+  clickNextButton() {
+    return this.form.clickSaveButton();
+  }
+
+  waitForErrorBox() {
+    return this.form.waitForErrorBox();
+  }
+
+  waitForIdentifierError(){
```
Comment (haishengwu-okta) [inline test/testcafe/framework/page-objects/IdentityPageObject.js]: why need both `waitForIdentifierError` and `waitForErrorBox`

File: test/testcafe/framework/page-objects/components/BaseFormObject.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+export default class BaseFormObject {
+  constructor (t, index) {
+    this.t = t;
+    this.form = new Selector('.o-form').nth(index || 0);
+  }
+
+  async setTextBoxValue(name, text) {
+    const element = this.form.find(`input[name="${name}"]`);
+    if (text === '') {
```
Comment (haishengwu-okta) [inline test/testcafe/framework/page-objects/components/BaseFormObject.js]: nit: 
`if (!text) {...`

File: test/testcafe/framework/page-objects/IdentityPageObject.js (hunk 3 above)
Comment (manueltanzi-okta) [reply test/testcafe/framework/page-objects/IdentityPageObject.js]: @haishengwu-okta You can have identifier error without box error (empty field on focus out)

**Pull request merged**